app/domain/authentication/authn_iam/authenticator.rb

```diff
@@ -0,0 +1,95 @@
+require 'net/http'
+require 'net/https'
+require 'json'
+
+module Authentication
+  module AuthnIam
+    class Authenticator
+      
+      def initialize(env:)
+        @env = env
+      end
+
+      def valid?(input)
+        @account, @login, password = input.account, input.username, input.password
+
+        # JSON holding the AWS signed headers 
+        signed_aws_headers = JSON.parse password
+
+        response_hash = identity_response_hash(signed_aws_headers)
+        trusted = response_hash != false
+
+        (trusted) && (iam_role_matches? host_role, response_hash)
+
+      end
+
+      def identity_response_hash(signed_aws_headers)
+
+        res = retrieve_iam_identity(signed_aws_headers)
+
+        Rails.logger.info("****> #{res.code} #{res.message}")
```

Comment:
It doesn't seem that it will be too clear in the log what this is referring to. So, if you want to provide diagnostic information to the server administrator, consider a single log message that contains all the info that someone might need. For example:

```
Rails.logger.info("Verification of IAM identity failed with HTTP code #{res.code.inspect}: #{res.message}")
```

Reply:
Fixed.